service/jmx/src/main/java/org/jolokia/service/jmx/handler/WriteHandler.java

```diff
@@ -132,21 +133,15 @@ private Object setAttribute(JolokiaWriteRequest request, MBeanServerConnection s
 
             // We set an inner value, hence we have to return provided value itself for resetting
             // it later back via JMX
-            return new Object[] {
-                    pCurrentValue,
-                    context.getMandatoryService(Serializer.class).setInnerValue(pCurrentValue, newValue, pathParts)
-            };
+            return context.getMandatoryService(Serializer.class).setInnerValue(pCurrentValue, newValue, pathParts);
```

Comment:
We could use here also a `new WriteRequestValues()` like in the else branch, so that `setInnerValue()` can still return the old value only. The reason why I don't like to put WriteRequestValues into the interface is that is (a) a new object, and (b) contains the original value `newValue` given as argument. People could be confused which to use later on because it is no **guaranteed** that both are the same. If you only have one (the method argument), this is unambiguous.